### ISSUE TITLE
Migrating from nose to nose2

### ISSUE THREAD

**Issue by ju2wheels** [nose-devs/nose2#248]
Hi all, I apologize for opening this here but I could not find any forums/IRC channels or other support/discussion areas for nose2.

Im trying to migrate a project that I have which was using nose to using nose2 with coverage reporting (branch coverage enabled). When using nose I get 100% coverage on my project but in migrating to nose2 I have run into issues which I have narrowed down to the way nose2 interacts with coverage and the apparent need for coverage specific configuration files with nose2 that I ddint need with nose before (as far as I can tell).

I have the following directory structure for my project:

```
.
├── python
│   ├── mainmodule
│   │   ├── api
│   │   │   ├── __init__.py
│   │   │   ├── mod1.py
│   │   │   ├── mod2.py
│   │   │   └── resource
│   │   │       ├── mod1.py
│   │   │       ├── formatter
│   │   │       │   ├── __init__.py
│   │   │       │   ├── mod1.py
│   │   │       │   └── mod2.py
│   │   │       ├── __init__.py
│   │   │       └── mod2.py
│   │   └── __init__.py
│   └── test
│       ├── __init__.py
│       └── unit
│           ├── __init__.py
│           └── mainmodule
│               ├── api
│               │   ├── __init__.py
│               │   ├── mod1_test.py
│               │   ├── mod2_test.py
│               │   └── resource
│               │       ├── mod1_test.py
│               │       ├── formatter
│               │       │   ├── __init__.py
│               │       │   ├── mod1_test.py
│               │       │   └── mod2_test.py
│               │       ├── __init__.py
│               │       └── mod2_test.py
│               └── __init__.py
├── setup.cfg
├── setup.py
├── unittest.cfg    # added for nose2, did not need for nose
├── .coveragerc   # added for nose2, did not need for nose
└── VERSION
```

With nose I am able to run the following and get 100% coverage with branch output:

```
python setup.py nosetests --cover-branches --cover-package mainmodule --rednose -s -v --with-coverage
```

It gives the report as:

```
Name                                         Stmts   Miss Branch BrMiss  Cover   Missing
----------------------------------------------------------------------------------------
mainmodule                                     2      0      0      0   100%   
mainmodule.api                                 0      0      0      0   100%   
mainmodule.api.mod1                           29      0     12      0   100%   
mainmodule.api.mod2                           79      0     18      0   100%   
mainmodule.api.resource                        0      0      0      0   100%   
mainmodule.api.resource.mod1                  60      0     20      0   100%   
mainmodule.api.resource.formatter              0      0      0      0   100%   
mainmodule.api.resource.formatter.mod1        18      0      8      0   100%   
mainmodule.api.resource.formatter.mod2        23      0     12      0   100%   
mainmodule.api.resource.mod2                  32      0      8      0   100%   
----------------------------------------------------------------------------------------
TOTAL                                          243      0     78      0   100%   

-----------------------------------------------------------------------------
31 tests run in 2.3 seconds (31 tests passed)
```

Using nose2 I had to add `unittest.cfg` and `.coveragerc` to even get a report close to the above but I still have problems with it reporting lines missing that are almost impossible to have been missed (pretty much identical to the issue [here](http://stackoverflow.com/questions/22146864/pytest-2-5-2-coverage-reports-missing-lines-which-must-have-been-processed)).

unittest.cfg:
```
[coverage]
always-on       = True
coverage-report = annotate
                  html
                  term
                  term-missing
                  xml

[unittest]
code-directories   = python
plugins            = nose2.plugins.coverage
                     nose2.plugins.junitxml
start-dir          = python
test-file-pattern  = *_test.py
test-method-prefix = test_
```

.coveragerc:
```
[run]
branch  = True
include = python/mainmodule/*
omit    = python/test/*
          test_*.*
source  = python
timid   = True
```

Running `nose2` produces the following report:
```
...............................
----------------------------------------------------------------------
Ran 31 tests in 1.970s

OK
---------- coverage: platform linux2, python 2.7.6-final-0 -----------
Name                                                Stmts   Miss Branch BrMiss  Cover   Missing
-----------------------------------------------------------------------------------------------
python/mainmodule/api/mod1                           29      7     12      0    83%   8-33, 73, 90
python/mainmodule/api/mod2                           79     26     18      0    73%   17-116, 169, 223
python/mainmodule/api/resource/mod1                  60     23     20      0    71%   9-38, 74, 103, 113-135, 150-221, 247
python/mainmodule/api/resource/formatter/mod1        18      3      8      0    88%   9-13
python/mainmodule/api/resource/formatter/mod2        23      4     12      0    89%   10-15
python/mainmodule/api/resource/mod2                  32     12      8      0    70%   9-34, 59-60, 105
-----------------------------------------------------------------------------------------------
TOTAL                                                 241     75     78      0    76%   
Coverage annotated source written next to source
Coverage HTML written to dir htmlcov
Coverage XML written to file coverage.xml
```

The misses here are all stuff that is almost impossible to miss (import/class/def lines mostly as shown in the stackoverflow link). The output format of the report is also using fully qualified paths as oppposed to the module names which I prefer better than the path and I have no idea why its like this.

My question is basically is there any documentation on how to migrate some of core nose coverage options into nose2 equivalents? The processes seem to be quite different and im not quite sure if I have it configured incorrectly or if Im missing config options.

**Comment by ztane** [comment]
Duplicate of https://github.com/nose-devs/nose2/issues/253

**Comment by ztane** [comment]
FWIW I switched to py.test and it worked right out of box for me as a nose2 replacement.

**Comment by little-dude** [comment]
This should be fixed in nose 0.6.0. Feel free to re-open if that's not the case.
